common/src/main/java/gov/cms/ab2d/common/service/JobServiceImpl.java

```diff
@@ -48,6 +61,25 @@ public Job createJob(String resourceTypes, String url) {
         job.setProgress(0);
         job.setUser(userService.getCurrentUser());
 
+        if (contractNumber != null) {
+            contractRepository.findContractByContractNumber(contractNumber).ifPresentOrElse(contractFound -> {
+                User user = userService.getCurrentUser();
+                Sponsor userSponsor = user.getSponsor();
+
+                List<Contract> contracts = userSponsor.getAggregatedAttestedContracts();
+                if (!contracts.contains(contractFound)) {
```

Comment:
Since `contains` method will check for object equality, we need to add both `equals` and `hashCode` to `Contract`. Let's do so using Lombok: the only field that should be included into equality check is `id`.